### PR TITLE
fix parsing for paths of format <hash>/path

### PR DESCRIPTION
**Pull request by whyrusleeping**
with this we can now do
```
ipfs object stat <hash>/a/b/c
```

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by wking** [comment]
On Tue, Jul 14, 2015 at 12:02:39PM -0700, Jeromy Johnson wrote:
> ipfs object stat <hash>/a/b/c

We could do that before (because len(parts) > 3).  This commit just
lets us do:

  $ ipfs object stat <hash>/a

Anyhow, the change itself (and tests :) look good to me.

File: test/sharness/t0051-object.sh [hunk 1]
```diff
@@ -114,6 +114,19 @@ test_object_cmd() {
 		test_cmp hwfile hwfile_out
 	'
 
+	test_expect_success "ipfs object stat path succeeds" '
+		ipfs object stat $(cat multi_patch)/a/b/c > obj_stat_out
+	'
```
Comment (jbenet) [inline test/sharness/t0051-object.sh]: maybe add this specific case too?

```
ipfs object stat $(cat multi_patch)/a > obj_stat_out_a
```

**Comment by jbenet** [comment]
LGTM. one comment above based on @wking's observation o/

File: test/sharness/t0051-object.sh (hunk 1 above)
Comment (whyrusleeping) [inline test/sharness/t0051-object.sh]: ah, true

**Comment by jbenet** [comment]
LGTM :+1:

**Pull request merged**